packages/@tinacms/cli/src/cmds/compile/index.ts

```diff
@@ -81,32 +80,46 @@ export const compile = async (_ctx, _next) => {
 
 const transpile = async (projectDir, tempDir) => {
   logger.info(logText('Transpiling...'))
-  // Make sure that post paths are posix (unix paths). This is necessary on windows.
-  const posixProjectDir = normalize(projectDir)
-  const posixTempDir = normalize(tempDir)
 
-  return Promise.all(
-    glob
-      // We will replaces \\ with / as required by docs see: https://github.com/mrmlnc/fast-glob#how-to-write-patterns-on-windows
-      .sync(path.join(projectDir, '**', '*.ts').replace(/\\/g, '/'), {
-        ignore: [
-          path
-            .join(projectDir, '__generated__', '**', '*.ts')
-            .replace(/\\/g, '/'),
-        ],
-      })
-      .map(async function (file) {
-        const fullPath = path.resolve(file)
-
-        const contents = await fs.readFileSync(fullPath).toString()
-        const newContent = ts.transpile(contents)
-        const newPath = file
-          .replace(posixProjectDir, posixTempDir)
-          .replace('.ts', '.js')
-        await fs.outputFile(newPath, newContent)
-        return true
-      })
+  const packageJSON = JSON.parse(
+    fs.readFileSync(packageJSONFilePath).toString() || '{}'
   )
+  const deps = packageJSON?.dependencies || []
+  const peerDeps = packageJSON?.peerDependencies || []
+  const devDeps = packageJSON?.devDependencies || []
+  const external = Object.keys({ ...deps, ...peerDeps, ...devDeps })
+
+  // Get file
+  const inputPathTS = path.join(projectDir, 'schema.ts')
+  const inputPathJS = path.join(projectDir, 'schema.js')
+  const inputPathTSX = path.join(projectDir, 'schema.tsx')
+  const inputPathJSX = path.join(projectDir, 'schema.jsx')
```

Comment:
This will be handy, one thing we may want to do in the near future is when `js` or `jsx` is provided, the generated SDK should be compiled to `js`.

Reply:
Yeah we would also have to compile the SDK to a separate file too. (it lives in type right now)